quickwit-search/src/collector.rs

```diff
@@ -342,7 +348,7 @@ fn merge_leaf_responses(
 ///
 /// TODO we could possibly optimize the sort away (but I doubt it matters).
 fn top_k_partial_hits(mut partial_hits: Vec<PartialHit>, num_hits: usize) -> Vec<PartialHit> {
-    partial_hits.sort_by(|left, right| {
+    partial_hits.sort_unstable_by(|left, right| {
```

Comment:
I wonder if we need `stable` sort here

Reply:
unstable is ok because partia_hit_sorting_key is a total order.